src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1390,4 +1391,43 @@ export class NetworkAreaDiagramViewer {
             }
         });
     }
+
+    public setJsonFlows(flows: string) {
+        const flowsArray: FLOW[] = JSON.parse(flows);
+        this.setFlows(flowsArray);
+    }
+
+    public setFlows(flows: FLOW[]) {
+        flows.forEach((flow) => {
+            this.setFlow(flow.branchId, flow.side, flow.p);
+        });
+    }
+
+    private setFlow(branchId: string, side: number, p: number) {
+        const edge: EdgeMetadata | undefined = this.diagramMetadata?.edges.find((edge) => edge.equipmentId == branchId);
```

Comment:
For a big network, we should probably have a map to access directly the edge with its equipmentId. 

Note that the same kind of direct access should be considered also for other edges.find() if we notice it's slow for a big network.

Reply:
Added a map for caching edges. The map could possibly be used also in other class methods